internal/operator/backrest/backup.go

```diff
@@ -293,7 +293,15 @@ func getCommandOptsFromPod(clientset *kubernetes.Clientset, task *crv1.Pgtask,
 	selector := fmt.Sprintf("%s=%s,%s in (%s,%s)", config.LABEL_PG_CLUSTER,
 		task.Spec.Parameters[config.LABEL_PG_CLUSTER], config.LABEL_PGHA_ROLE,
 		"promoted", config.LABEL_PGHA_ROLE_PRIMARY)
-	pods, err := kubeapi.GetPods(clientset, selector, namespace)
+
+	options := meta_v1.ListOptions{
+		FieldSelector: `status.phase=Running`,
```

Comment:
We might want to consider using the [fields package](https://godoc.org/k8s.io/apimachinery/pkg/fields) for creating our selectors, e.g.:

```go
FieldSelector: fields.OneTermEqualSelector("status.phase", "Running").String()
```

There are also [constants we can leverage for the specific pod phases](https://godoc.org/k8s.io/api/core/v1#PodPhase)  allowing us to then do:

```go

import(
    corev1 "k8s.io/api/core/v1"
)

FieldSelector: fields.OneTermEqualSelector("status.phase", string(corev1.PodRunning)).String()
```

Reply:
Yeah, that seems better.